core/src/mender-deployment-data.c

```diff
@@ -59,6 +59,7 @@ validate_deployment_data(const cJSON *deployment_data) {
         { .key = MENDER_DEPLOYMENT_DATA_KEY_PROVIDES, .type = cJSON_IsString },               /* Artifact provides (filtered on clears provides) */
         /* { .key = MENDER_DEPLOYMENT_DATA_KEY_STATE, .type = cJSON_IsString }, */            /* TODO: MEN-7515: State name */
         { .key = MENDER_DEPLOYMENT_DATA_KEY_STATE_DATA_STORE_COUNT, .type = cJSON_IsNumber }, /* State data store count */
+        { .key = MENDER_DEPLOYMENT_DATA_KEY_ROLLBACK_REBOOT_COUNT, .type = cJSON_IsNumber },  /* Rollback reboot count */
```

Comment:
I don't think we need to save this counter into the non-volatile storage.

It will retry up to the configured times during the current boot, and on spontaneous reboots it will retry from scratch (which might be suboptimal) but there is no risk of it getting stuck forever because we have the overall "State data store count".

Is there an extra benefit that am I missing? My motivation for not adding it is to keep the state data as simple as possible (it is already too complex IMO).

Reply:
> Is there an extra benefit that am I missing?
I don't think so. I didn't think of the state data store count, so I thought it could get stuck. 
After looking at the code again, I agree that we don't need to store the counter

Reply:
Discussed in person, we can drop the counter altogether.